From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Axel Garcia K." <git@axelgk.com>
Date: Wed, 6 May 2026 09:11:38 +0000
Subject: [PATCH 4/4] Allow external PicoDrive media support

Add EXTERNAL_CHD_MEDIA and EXTERNAL_ZLIB build knobs so the SF2000 build can keep PicoDrive CD/CHD and gzip-backed media support enabled while linking libchdr, zstd, lzma, and zlib from a shared archive.
---
 Makefile | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/Makefile b/Makefile
index 82a740582528844ce030cbe0ea008b544a047423..9c3e0ed4f2ea54c32ce0d69dd4e38b0eb68a408e 100644
--- a/Makefile
+++ b/Makefile
@@ -350,6 +350,9 @@ ifneq "$(NO_CD_MEDIA)" "1"
 ifeq (1,$(use_libchdr))
 CFLAGS += -DUSE_LIBCHDR
 
+ifeq (1,$(EXTERNAL_CHD_MEDIA))
+CFLAGS += $(EXTERNAL_CHD_INCFLAGS) $(EXTERNAL_ZLIB_INCFLAGS)
+else
 # chdr
 CHDR = pico/cd/libchdr
 CHDR_OBJS += $(CHDR)/src/libchdr_chd.o $(CHDR)/src/libchdr_cdrom.o
@@ -383,14 +386,17 @@ endif
 CFLAGS := -I$(LZMA)/include -I$(CHDR)/include $(CFLAGS)
 endif
 endif
+endif
 
 ifeq "$(PLATFORM_ZLIB)" "1"
 # zlib
 ZLIB_OBJS = zlib/gzio.o zlib/inffast.o zlib/inflate.o zlib/inftrees.o zlib/trees.o \
 	zlib/deflate.o zlib/crc32.o zlib/adler32.o zlib/zutil.o zlib/compress.o zlib/uncompr.o
 CFLAGS += -Izlib
+ifneq (1,$(EXTERNAL_ZLIB))
 OBJS += $(ZLIB_OBJS)
 endif
+endif
 # unzip
 OBJS += unzip/unzip.o
 
-- 
2.54.0

